### DISCUSSION TITLE
Allow additional arguments on `Filament\authorize()` to be passed on `Laravel\authorize()`

### DISCUSSION THREAD

**Discussion by juliomotol** [filamentphp/filament#14130] [Ideas]
Given the ff policy:

```php
class PostPolicy
{
    public function viewAny(User $user, PostTypeEnum $type): bool
    {
        // ...
    }
}
```

In our case, we have different Resources for each type of Post (don't ask me why, I suggested to just use filters but the client insisted on separating each type), and certain users may only have access on certain types of Posts

At the moment, Filament's authorize method is unable to pass a $type argument to the policy method unlike in Laravel's authorize method.

See #14121 for more info.

**Comment by saeedvaziry** [comment]
Hey @danharrin 

I was going to propose the same idea and then searched and found this.

I also faced the same limit when I was trying to authorize more complex policies in my app. Sometimes you need to pass more arguments to your Policy to authorize an action.

By default, Laravel allows this by passing the arguments as an array instead of the model object itself.

I think it would be a good addition to Filament as well.

**Comment by ryanmortier** [comment]
Hey @danharrin 

I just found this after searching because I too would like to pass additional arguments to the authorize.

My use case is the following:

I have a resource for `WorkOrder` and a relation manager for `WorkOrderNote`.

Currently the way Filament works is that if I want to create a work order note, it checks a `create` method on the `WorkOrderNotePolicy` class which is fine but if I have a permission structure where I want basic users to only be able to create notes on work orders they own, I need to be able to pass the work order itself (the owner record) to the create method on the `WorkOrderNotePolicy` class.

I have the following workaround but it'd be nice if the built in `authorize` function allowed this.

```php
    protected function canCreate(): bool
    {
        return Gate::allows('create', [$this->getTable()->getModel(), $this->ownerRecord]);
    }
```

**Comment by danharrin** [comment]
The idea of the `authorise()` method is to dynamically determine the policy based on its existence and all together wire everything up automatically. It is an internal function and not documented. If you want to customise the permission check, you should use `user->can()` or `Gate::allows()`, since you already know for sure that the policy exists etc. Using the `authorize()` method in this case is not appropriate.
